src/conn.c

```diff
@@ -3797,3 +3914,38 @@ natsConn_signatureHandler(char **customErrTxt, unsigned char **psig, int *sigLen
     return NATS_UPDATE_ERR_STACK(s);
 }
 
+natsStatus
+natsConn_signatureHandler(char **customErrTxt, unsigned char **psig, int *sigLen, const char *nonce, void *closure)
+{
+    natsStatus      s       = NATS_OK;
+    userCreds       *uc     = (userCreds*) closure;
+
+    s = _sign(uc, (const unsigned char*) nonce, 0, psig, sigLen);
+    return NATS_UPDATE_ERR_STACK(s);
+}
+
+natsStatus
+natsConnection_Sign(natsConnection *nc, const unsigned char *payload, int payloadLen, unsigned char **sig, int *sigLen)
```

Comment:
sigLen should be 64bytes yes? Should we check it?

Reply:
The function allocates, but then yes, I could remove this param since the returned signature will be 64 bytes.

Reply:
Agree, but why do we not pass in addr to a 64 byte char array? That way we can stick it on the stack if we want.

Reply:
Because low level crypto_sign() that I use has the signature followed by original input in same memory block. So I need to allocate (see natsKeys_Sign() and alloc of `sm`).
Now of course, I could still have the alloc and then make another copy to give byte array, but that's an extra memcopy..

Reply:
You can allocate but public function should take *char to a 64byte block that we fill in.

Reply:
will do